### ISSUE TITLE
How does superagent handle ports in URLs?

### ISSUE THREAD

**Issue by eliprand** [visionmedia/superagent#1241]
I have the following sample code (in typescript, sorry) using _superagent_ on NodeJS:

```typescript
import * as agent from 'superagent';
import * as http from 'http';


describe('superagent', () => {
    const url = 'http://localhost:3002/';
    it('should call localhost', done => {
        agent.get(url)
            .then(_ => {
                console.log(_.body);
                done();
            })
            .catch(done);
    });

    it('should call localhost (http)', done => {
        http.get(url, res => {
            if (res.statusCode !== 200) {
                res.resume();
                done(res);
            } else {
                res.on('data', _ => {
                    console.log(_.toString());
                });
                res.on('end', done);
            }

        });
    });
});
```
I get the following output:

```text
> tsc && mocha index.js


  superagent
{}
    ✓ should call localhost
Hello World
    ✓ should call localhost (http)

  2 passing (195ms)
```

The `http` call is obviously correct, but it seems that _superagent_ just calls something else entirely. What am I missing? How does one specify a port to a _superagent_ request?

**Comment by focusaurus** [comment]
Probably more precise to log `_.text` with the superagent callback. The `body` property is used when we parse response JSON into an object. In your case you test service is probably just sending `text/plain` with `Hello World` which will be available via superagent in `_.text` (`_` being the response object, more typically named `res`).

I don't think ports have anything to do with this - just understanding the superagent response API property semantics.

**Comment by eliprand** [comment]
@focusaurus thanks for taking the time to looking at the post. You are correct. This was a very contrived example that was exhibiting the same symptoms as my more complex example and I thought I had found a smoking gun.

My symptoms were that the requests were getting a 404 response and not even hitting my handler (in express, debugger attached).

As it turns out, I had left a VM running last week that was handling the request instead. Now, why would requests made within NodeJS (using superagent or http or request) were being routed to the VM while Postman requests were being routed properly? I don't know. I turned off the VM and everything is working properly.
